xs2a-adapter-service-remote/src/main/java/de/adorsys/xs2a/adapter/service/impl/AccountInformationServiceImpl.java

```diff
@@ -0,0 +1,203 @@
+        Map<String, String> requestParamMap = requestParams.toMap();
+        LocalDate dateFrom = strToLocalDate(requestParamMap.get(RequestParams.DATE_FROM));
+        LocalDate dateTo = strToLocalDate(requestParamMap.get(RequestParams.DATE_TO));
+        String entryReferenceFrom = requestParamMap.get(RequestParams.ENTRY_REFERENCE_FROM);
+        BookingStatusTO bookingStatus = BookingStatusTO.fromValue(requestParamMap.get(RequestParams.BOOKING_STATUS));
+        Boolean deltaList = Boolean.valueOf(requestParamMap.get(RequestParams.DELTA_LIST));
+        Boolean withBalance = Boolean.valueOf(requestParamMap.get(RequestParams.WITH_BALANCE));
+        return client.getTransactionList(s, dateFrom, dateTo, entryReferenceFrom, bookingStatus, deltaList, withBalance, headersMap);
+    }
+
+    @Override
+    public GeneralResponse<String> getTransactionListAsString(String s, RequestHeaders requestHeaders, RequestParams requestParams) {
+        Map<String, String> headersMap = requestHeaders.toMap();
+        ResponseEntity<Object> responseEntity = getTransactionListFromClient(s, requestParams, headersMap);
+
+        return new GeneralResponse<>(responseEntity.getStatusCodeValue(), (String) responseEntity.getBody(), getHeaders(responseEntity.getHeaders()));
+    }
+
+    @Override
+    public GeneralResponse<ScaStatusResponse> getConsentScaStatus(String s, String s1, RequestHeaders requestHeaders) {
+        ResponseEntity<ScaStatusResponseTO> responseEntity = client.getConsentScaStatus(s, s1, requestHeaders.toMap());
+        ScaStatusResponse scaStatusResponse = scaStatusResponseMapper.toScaStatusResponse(responseEntity.getBody());
+        return new GeneralResponse<>(responseEntity.getStatusCodeValue(), scaStatusResponse, getHeaders(responseEntity.getHeaders()));
+    }
+
+    @Override
+    public GeneralResponse<BalanceReport> getBalances(String s, RequestHeaders requestHeaders) {
+        ResponseEntity<ReadAccountBalanceResponse200TO> responseEntity = client.getBalances(s, requestHeaders.toMap());
+        BalanceReport balanceReport = balanceReportMapper.toBalanceReport(responseEntity.getBody());
+        return new GeneralResponse<>(responseEntity.getStatusCodeValue(), balanceReport, getHeaders(responseEntity.getHeaders()));
+    }
+
+    private ResponseHeaders getHeaders(HttpHeaders httpHeaders) {
+        Set<Map.Entry<String, List<String>>> entrySet = httpHeaders.entrySet();
+        Map<String, String> headers = new HashMap<>(entrySet.size());
+        for (Map.Entry<String, List<String>> entry : entrySet) {
+            List<String> value = entry.getValue();
+            if (value != null && !value.isEmpty()) {
+                headers.put(entry.getKey(), value.get(0));
+            }
+        }
+        return ResponseHeaders.fromMap(headers);
+    }
+
+    private LocalDate strToLocalDate(String date) {
+        if (StringUtils.isNotBlank(date)) {
+            return LocalDate.parse(date);
```

Comment:
Is it a correct behaviour for us: if `String date` has incorrect format -> `LocalDate.parse(date)` throws `DateTimeParseException` -> user gets `500` response?

Reply:
added parse exception and handler in the example app